build.gradle

```diff
@@ -213,53 +217,45 @@ publishing {
                             name 'Andreas Marek'
                         }
                     }
+                    scm {
+                        url "https://github.com/atlassian-labs/nadel"
+                    }
                 }
             }
         }
     }
 }
 
+artifactory {
+    publish {
+        contextUrl = 'https://packages.atlassian.com/'
 
-bintray {
-    user = System.env.BINTRAY_USER ?: project["bintray.user"]
-    key = System.env.BINTRAY_API_KEY ?: project["bintray.key"]
-    publications = ['graphqlJava']
-    publish = true
-    pkg {
-        repo = 'nadel'
-        name = 'nadel'
-        desc = 'GraphQL stitching DSL'
-        licenses = ['MIT']
-        vcsUrl = 'https://github.com/graphql-java/nadel'
-
-        version {
-            name = project.version
-            desc = project.description
-            released = new Date()
-            vcsTag = 'v' + project.version
-            gpg {
-                sign = true
-            }
-            mavenCentralSync {
-                sync = true
-                user = System.env.MAVEN_CENTRAL_USER
-                password = System.env.MAVEN_CENTRAL_PASSWORD
-                close = '1'
-            }
+        repository {
+            repoKey = 'maven-central'
+            username = System.env.ARTIFACTORY_USERNAME
+            password = System.env.ARTIFACTORY_API_KEY
+        }
+        defaults {
+            publications('maven')
+            publishIvy = false
+            // This needs to be "false", otherwise, the artifactory plugin will try to publish
+            // a build info file to Artifactory and fail because we don't have the permissions to do that.
+            publishBuildInfo = false
         }
     }
 }
 
+signing {
+    useInMemoryPgpKeys("$System.env.SIGNING_KEY", "$System.env.SIGNING_PASSWORD")
+    sign publishing.publications.graphqlJava
+}
+
 // all publish tasks depend on the build task
 tasks.withType(PublishToMavenRepository) {
     dependsOn build
 }
 
-
 task myWrapper(type: Wrapper) {
-    gradleVersion = '4.10.2'
-    distributionUrl = "https://services.gradle.org/distributions/gradle-${gradleVersion}-all.zip"
+    gradleVersion = '5.6.4'
```

Comment:
we should update to gradle 6.6 or so ... have a look at graphql-java

Reply:
run this twice

```
./gradlew wrapper --gradle-version 6.6.1
./gradlew wrapper --gradle-version 6.6.1
```
This will upgrade the version